src/cls/rgw/cls_rgw_types.h

```diff
@@ -340,6 +340,10 @@ struct cls_rgw_obj_key {
   cls_rgw_obj_key(const std::string &_name) : name(_name) {}
   cls_rgw_obj_key(const std::string& n, const std::string& i) : name(n), instance(i) {}
 
+  operator std::string() const {
```

Comment:
(oops, this comment didn't submit with the approval)

in general i'd prefer explicit conversions (like a `to_string()` function) over implicit ones, which would allow you to accidentally pass a key to a function/overload taking string

ideally this formatting would be in a stream output `operator<<`, but that doesn't help with the printf-style `CLS_LOG()`

Reply:
Fair point, I fixed it. And I added operator<< also. But yeah, to create a std::streamstring just for the logging would be a pain.